test/apps/iodemo/ucx_wrapper.h

```diff
@@ -308,6 +310,16 @@ class UcxConnection {
         return _conn_id;
     }
 
+    uint64_t remote_id() const {
+        return _remote_conn_id;
+    }
+
+    void set_remote_id(uint64_t remote_conn_id) {
+        assert(_remote_conn_id == std::numeric_limits<uint64_t>::max());
+        assert(_use_am);
```

Comment:
if it's relevant only for `_use_am` better to check it outside

Reply:
yes, since we set remote ID for TAG through STREAM send/recv, the function should be called for AM only.
better to keep the check here to be protected if someone move it to another place